services/identity/MailClient.ts

```diff
@@ -2,14 +2,23 @@ import axios from "axios"
 
 import logger from "@logger/logger"
 
-const { POSTMAN_API_KEY } = process.env
 const POSTMAN_API_URL = "https://api.postman.gov.sg/v1"
 
 class MailClient {
-  async sendMail(recipient: string, body: string): Promise<void> {
-    if (!POSTMAN_API_KEY)
+  // NOTE: This is set as a private readonly property
+  // rather than a variable within the file for testing.
+  // This is to allow us to test that initialization fails when the API key is empty
+  private readonly POSTMAN_API_KEY: string
+
+  constructor() {
+    const { POSTMAN_API_KEY } = process.env
+    if (!POSTMAN_API_KEY) {
       throw new Error("Postman.gov.sg API key cannot be empty.")
+    }
+    this.POSTMAN_API_KEY = POSTMAN_API_KEY
+  }
 
+  async sendMail(recipient: string, body: string): Promise<void> {
```

Comment:
I feel that changing this structure to support tests for the presence of the env var makes our code less readable, could we look into changing the env var directly in the test instead? 

See https://stackoverflow.com/a/48042799

Reply:
i do think it'll be better but i'm not sure this can be done - typescript complains about this: `An import declaration can only be used at the top level of a namespace or module.`, which would seem to imply that dynamic imports (in the test itself) wouldn't be possible

Reply:
fixed - typescript has dynamic imports using `const x = await (import(your_path_here))` but take note that this requires that it is either 1. destructured or 2. called using `.default`like so: `const x =  await (import(your_path_here)).default`